File: lemongrass/src/main/java/com/ttoannguyen/lemongrass/dto/request/recipe/RecipeUpdateRequest.java
package com.ttoannguyen.lemongrass.dto.Request.recipe;

import com.ttoannguyen.lemongrass.dto.Request.image.ImageRequest;
import com.ttoannguyen.lemongrass.dto.Request.ingredient.IngredientCreationRequest;
import com.ttoannguyen.lemongrass.dto.Request.instruction.InstructionCreationRequest;
import com.ttoannguyen.lemongrass.dto.Request.tag.TagCreationRequest;
import com.ttoannguyen.lemongrass.entity.enums.Difficulty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class RecipeUpdateRequest {
  private String id;
  private String title;
  private Integer cookingTime;
  private Difficulty difficulty;

  private String description;
  private Integer servings;
  private List<String> categoryIds;
  private List<TagCreationRequest> tags;
  private List<IngredientCreationRequest> ingredients;
  private List<InstructionCreationRequest> instructions;
  private List<ImageRequest> images;
}

File: lemongrass-client/src/types/Recipe/RecipeUpdateRequest.ts
import { Difficulty } from "@/types/enums/difficulty.enum";

export type RecipeIngredientUpdateRequest = {
  id?: string;
  templateId: string;
  unitId: string;
  quantity: number;
  note?: string;
  orderIndex: number;
};

export type RecipeInstructionUpdateRequest = {
  id?: string;
  stepNumber: number;
  description: string;
  images: ImageUpdateUpload[];
};

export type ImageUpdateUpload = {
  id?: string;
  previewUrl?: string;
  file?: File;
  displayOrder: number;
};

export type RecipeUpdateRequest = {
  id: string;
  title: string;
  description: string;
  cookingTime: number;
  servings: number;
  difficulty: Difficulty;
  categoryIds: string[];
  ingredients: RecipeIngredientUpdateRequest[];
  instructions: RecipeInstructionUpdateRequest[];
  images: ImageUpdateUpload[];
};
